databricks/koalas/tests/test_dataframe.py

```diff
@@ -580,23 +580,48 @@ def test_append(self):
             kdf.append(kdf, sort=True)
 
         # Assert using 'verify_integrity' only raises an exception for overlapping indices
-        kdf.append(other_kdf, verify_integrity=True)
+        self.assert_eq(kdf.append(other_kdf, verify_integrity=True),
+                       pdf.append(other_pdf, verify_integrity=True))
         msg = "Indices have overlapping values"
         with self.assertRaises(ValueError, msg=msg):
             kdf.append(kdf, verify_integrity=True)
 
+        # Skip integrity verification when ignore_index=True
+        self.assert_eq(kdf.append(kdf, ignore_index=True, verify_integrity=True),
+                       pdf.append(pdf, ignore_index=True, verify_integrity=True))
+
         # Assert appending multi-index DataFrames
         multi_index_pdf = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'),
                                        index=[[2, 3], [4, 5]])
         multi_index_kdf = ks.from_pandas(multi_index_pdf)
+        other_multi_index_pdf = pd.DataFrame([[5, 6], [7, 8]], columns=list('AB'),
+                                             index=[[2, 3], [6, 7]])
+        other_multi_index_kdf = ks.from_pandas(other_multi_index_pdf)
+
```

Comment:
Since you already define `other_multi_index_pdf` here, maybe you could also call `self.assert_eq(multi_index_kdf.append(other_multi_index_kdf, verify_integrity=True),
                       multi_index_pdf.append(other_multi_index_pdf, verify_integrity=True))` in the assertion below. This would however be more for the sake of consistency and shouldn't make any technical difference.

Otherwise, the PR looks fine to me.

Reply:
Do you mean lines 605-606 below?

Reply:
No, I meant replacing lines 601 and 602 with what I wrote.

Reply:
What you wrote is the same as lines 605-606, right?

Lines 601-602 is testing different thing as you did at line 566, whether we can append the dataframes with duplicate indexes if verify_integrity is False (and the result is the same as pandas').
Or do you want to add the same check as line 570, like `self.assert_eq(multi_index_kdf.append(other_multi_index_kdf), multi_index_pdf.append(other_multi_index_pdf))`?

Reply:
Oh sorry, my bad. I meant without `verify_integrity=True`. So instead of appending `multi_index_kdf` to itself, you might just do `multi_index_kdf.append(other_multi_index_kdf)` since you've already defined `other_multi_index_kdf` at that point anyway. Sorry again for the confusion.

Reply:
I see, thanks! I added the test (513f339), so I think this is ready to merge now.